### PR TITLE
Switch to Distube's ytdl-core fork (resolves playback issue)

### LINKED ISSUE #1037: Unable to play any songs
**What's happening?**
I am currently getting an error when trying to play any songs, the bot says "ope: No playing song found".

**Logs**

```
 +0ms
  prisma:client:libraryEngine  sending request, this.libraryStarted: true +6ms
  prisma:client  Prisma Client call: +6ms
  prisma:client  prisma.fileCache.findUnique({
  where: {
    hash: 'afd225007a8561b11ff551d42cd062fead6b66e44d920fc0b2f6a45c37871fce1b7539264a0409f3076aa27ecffb54de11d2466a86c939714632e50c9d55c7ae'
  }
}) +0ms
  prisma:client  Generated request: +0ms
  prisma:client  query {
  findUniqueFileCache(where: {
    hash: "afd225007a8561b11ff551d42cd062fead6b66e44d920fc0b2f6a45c37871fce1b7539264a0409f3076aa27ecffb54de11d2466a86c939714632e50c9d55c7ae"
  }) {
    hash
    bytes
    accessedAt
    createdAt
    updatedAt
  }
}
 +0ms
  prisma:client:libraryEngine  sending request, this.libraryStarted: true +6ms
  muse [VOICE] received voice state update: {"member":{"user":{"username":"SwoftBot","public_flags":0,"id":"985127745015128084","global_name":null,"display_name":null,"discriminator":"3140","clan":null,"bot":true,"avatar_decoration_data":null,"avatar":"9eb046c7d270f2fa7e83394af0e2b30c"},"roles":["987505496963555339"],"premium_since":null,"pending":false,"nick":null,"mute":false,"joined_at":"2022-06-17T23:54:17.468000+00:00","flags":0,"deaf":false,"communication_disabled_until":null,"avatar":null},"user_id":"985127745015128084","suppress":false,"session_id":"b8c909325b5757be574d3d2486b4988a","self_video":false,"self_mute":false,"self_deaf":true,"request_to_speak_timestamp":null,"mute":false,"guild_id":"136921565056073728","deaf":false,"channel_id":"505091962995474446"} +109ms
  prisma:client  Prisma Client call: +101ms
  prisma:client  prisma.setting.findUnique({
  where: {
    guildId: '136921565056073728'
  }
}) +1ms
  prisma:client  Generated request: +0ms
  prisma:client  query {
  findUniqueSetting(where: {
    guildId: "136921565056073728"
  }) {
    guildId
    playlistLimit
    secondsToWaitAfterQueueEmpties
    leaveIfNoListeners
    queueAddResponseEphemeral
    autoAnnounceNextSong
    defaultVolume
    createdAt
    updatedAt
  }
}
 +0ms
  prisma:client:libraryEngine  sending request, this.libraryStarted: true +103ms
  muse [VOICE] received voice server: {"t":"VOICE_SERVER_UPDATE","s":8,"op":0,"d":{"token":"621e7999a1fccdaa","guild_id":"136921565056073728","endpoint":"us-east7654.discord.media:443"}} +25ms
  prisma:client  Prisma Client call: +1s
  prisma:client  prisma.setting.findUnique({
  where: {
    guildId: '136921565056073728'
  }
}) +0ms
  prisma:client  Generated request: +1ms
  prisma:client  query {
  findUniqueSetting(where: {
    guildId: "136921565056073728"
  }) {
    guildId
    playlistLimit
    secondsToWaitAfterQueueEmpties
    leaveIfNoListeners
    queueAddResponseEphemeral
    autoAnnounceNextSong
    defaultVolume
    createdAt
    updatedAt
  }
}
 +0ms
  prisma:client:libraryEngine  sending request, this.libraryStarted: true +1s
  muse Rick Astley - Never Gonna Give You Up (Official Music Video) is unavailable +1s
  muse Error: No playing song found
  muse     at buildPlayingMessageEmbed (file:///opt/bot/src/utils/build-embed.ts:1:1533)
  muse     at AddQueryToQueue.addToQueue (file:///opt/bot/src/services/add-query-to-queue.ts:1:3628)
  muse     at async play_default.execute (file:///opt/bot/src/commands/play.ts:1:1604)
  muse     at async Client.<anonymous> (file:///opt/bot/src/bot.ts:1:2354) +1ms

```

**Screenshots**
image

**Runtime**
I'm running Muse:
- [x] Directly from the cloned repository
- [ ] Inside a Docker container
- [ ] Something else (please elaborate)

**Versions**
- Muse: v2.8.1
- Docker (if applicable): N/A
- OS: Debian GNU/Linux 12
- Node.js: v18.19.0
- ffmpeg: ffmpeg version 5.1.4-0+deb12u1

### PR DESCRIPTION
**Pull request by MarcoCoreDuo**
Closes #1037 

I don't think ytdl-core will receive any updates anymore and play-dl also has this playback issue. Distube's fork has some extra fixes and also updated decipher code (resolves the playback issue). It works as a drop-in replacement. The only downside is that it needs nodejs >= 18.

- [x] I updated the changelog

### CODE REVIEW

**Comment by EugeneLeclerc** [comment]
Well I just did the same, you were 30 minutes faster than me: https://github.com/codetheweb/muse/pull/1044

While we are updating Node, I think we could go directly to Node 20? It is the current LTS and has been for quite a while already.

**Comment by MarcoCoreDuo** [comment]
> Well I just did the same, you were 30 minutes faster than me: #1044
> 
> While we are updating Node, I think we could go directly to Node 20? It is the current LTS and has been for quite a while already.

I'm not really sure if we could use Node 20 without updating dependencies. I am glad that the current Dockerfile now works (Debian-Bookworm had a too-new libc-bin). But I can try building the Dockerfile with Node 20.

**Comment by EugeneLeclerc** [comment]
> I'm not really sure if we could use Node 20 without updating dependencies. I am glad that the current Dockerfile now works (Debian-Bookworm had a too-new libc-bin). But I can try building the Dockerfile with Node 20.
Yeah getting things working in the priority I guess, Node 20 can wait 😄

**Comment by MarcoCoreDuo** [comment]
> > I'm not really sure if we could use Node 20 without updating dependencies. I am glad that the current Dockerfile now works (Debian-Bookworm had a too-new libc-bin). But I can try building the Dockerfile with Node 20.
> > Yeah getting things working in the priority I guess, Node 20 can wait 😄

I tried building the image with Node 20, and it does not work because there are no pre-built binaries for @discordjs/opus. So Node 20 would require Python and build tools in the image.

File: package.json [hunk 1]
```diff
@@ -89,6 +89,7 @@
     "@discordjs/opus": "^0.8.0",
     "@discordjs/rest": "1.0.1",
     "@discordjs/voice": "0.11.0",
+    "@distube/ytdl-core": "^4.13.5",
```
Comment (jamcdon) [inline package.json:92]: Could fix current bug in bot

**Review by jamcdon — APPROVED** [review]

**Comment by Stealthii** [comment]
Using the suggested `codetheweb/muse:pr-1042` image, I've seen the issue resolved, and not encountered any other warnings or issues during a few hours testing.

**Comment by 2KDrop** [comment]
Did further testing, seems to happen only when skipping the last song in the queue or skipping to the last song in the queue, have tested 3 videos and gotten the same results from all of them.

Old Gods of Asgard's Herald of Darkness audio track - https://www.youtube.com/watch?v=TyC1NzSmFIg
Noah's Face to Face audio track - https://www.youtube.com/watch?v=oLcoljNKh3I
Fart with reverb sound effect - https://www.youtube.com/watch?v=9FLRHejWAo8

All play through fine but upon skipping to the last or the last video itself the bot crashes with the error below.

/usr/app/src/services/player.ts:315
      throw new Error('No songs in queue to forward to.');
            ^


Error: No songs in queue to forward to.
    at default.manualForward (/usr/app/src/services/player.ts:315:13)
    at default.forward (/usr/app/src/services/player.ts:277:10)
    at default.onAudioPlayerIdle (/usr/app/src/services/player.ts:586:18)
    at AudioPlayer.emit (node:events:517:28)
    at AudioPlayer.set state (/usr/app/node_modules/@discordjs/voice/src/audio/AudioPlayer.ts:356:9)
    at AudioPlayer.stop (/usr/app/node_modules/@discordjs/voice/src/audio/AudioPlayer.ts:493:9)
    at default.forward (/usr/app/src/services/player.ts:283:27)
    at default_1.execute (/usr/app/src/commands/skip.ts:37:20)
    at Client.<anonymous> (/usr/app/src/bot.ts:79:27)
    at Client.emit (node:events:517:28)

Node.js v18.20.4

I've attached a screenshot to showcase an example of what to do to recreate the bug
image

**Comment by MarcoCoreDuo** [comment]
> Did further testing, seems to happen only when skipping the last song in the queue or skipping to the last song in the queue, have tested 3 videos and gotten the same results from all of them.
> 
> Old Gods of Asgard's Herald of Darkness audio track - https://www.youtube.com/watch?v=TyC1NzSmFIg Noah's Face to Face audio track - https://www.youtube.com/watch?v=oLcoljNKh3I Fart with reverb sound effect - https://www.youtube.com/watch?v=9FLRHejWAo8
> 
> All play through fine but upon skipping to the last or the last video itself the bot crashes with the error below.
> 
> /usr/app/src/services/player.ts:315 throw new Error('No songs in queue to forward to.'); ^
> 
> Error: No songs in queue to forward to. at default.manualForward (/usr/app/src/services/player.ts:315:13) at default.forward (/usr/app/src/services/player.ts:277:10) at default.onAudioPlayerIdle (/usr/app/src/services/player.ts:586:18) at AudioPlayer.emit (node:events:517:28) at AudioPlayer.set state (/usr/app/node_modules/@discordjs/voice/src/audio/AudioPlayer.ts:356:9) at AudioPlayer.stop (/usr/app/node_modules/@discordjs/voice/src/audio/AudioPlayer.ts:493:9) at default.forward (/usr/app/src/services/player.ts:283:27) at default_1.execute (/usr/app/src/commands/skip.ts:37:20) at Client. (/usr/app/src/bot.ts:79:27) at Client.emit (node:events:517:28)
> 
> Node.js v18.20.4
> 
> I've attached a screenshot to showcase an example of what to do to recreate the bug image

Thank you for reporting. This was an error from a different pr from me. This also resulted in the bot always skipping an additional song. Should now be fixed.

**Comment by Stealthii** [comment]
> Using the suggested `codetheweb/muse:pr-1042` image, I've seen the issue resolved, and not encountered any other warnings or issues during a few hours testing.

Spoke too soon, got these exceptions today:
```
tmp-muse-1  | Error: type lockupViewModel is not known
tmp-muse-1  |     at parseItem (/usr/app/node_modules/ytsr/lib/parseItem.js:83:13)
tmp-muse-1  |     at catchAndLogFunc (/usr/app/node_modules/ytsr/lib/parseItem.js:90:12)
tmp-muse-1  |     at module.exports (/usr/app/node_modules/ytsr/lib/parseItem.js:119:46)
tmp-muse-1  |     at /usr/app/node_modules/ytsr/lib/main.js:53:34
tmp-muse-1  |     at Array.map (<anonymous>)
tmp-muse-1  |     at module.exports (/usr/app/node_modules/ytsr/lib/main.js:53:25)
tmp-muse-1  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
tmp-muse-1  |     at KeyValueCacheProvider.wrap (/usr/app/src/services/key-value-cache.ts:53:20)
tmp-muse-1  |     at async run (file:///usr/app/node_modules/p-queue/dist/index.js:109:36)
tmp-muse-1  |
tmp-muse-1  | /********************************************************************************************************************************************************************************************************
tmp-muse-1  | failed at func parseItem: type lockupViewModel is not known
tmp-muse-1  | pls post the the files in /usr/app/node_modules/ytsr/dumps to https://github.com/TimeForANinja/node-ytsr/issues
tmp-muse-1  | os: linux-x64, node.js: v18.20.4, ytsr: 3.8.4
```

I've just pulled the updates from 3c161b77 and will test today.

**Comment by MarcoCoreDuo** [comment]
> > Using the suggested `codetheweb/muse:pr-1042` image, I've seen the issue resolved, and not encountered any other warnings or issues during a few hours testing.
> 
> Spoke too soon, got these exceptions today:
> 
> ```
> tmp-muse-1  | Error: type lockupViewModel is not known
> tmp-muse-1  |     at parseItem (/usr/app/node_modules/ytsr/lib/parseItem.js:83:13)
> tmp-muse-1  |     at catchAndLogFunc (/usr/app/node_modules/ytsr/lib/parseItem.js:90:12)
> tmp-muse-1  |     at module.exports (/usr/app/node_modules/ytsr/lib/parseItem.js:119:46)
> tmp-muse-1  |     at /usr/app/node_modules/ytsr/lib/main.js:53:34
> tmp-muse-1  |     at Array.map (<anonymous>)
> tmp-muse-1  |     at module.exports (/usr/app/node_modules/ytsr/lib/main.js:53:25)
> tmp-muse-1  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> tmp-muse-1  |     at KeyValueCacheProvider.wrap (/usr/app/src/services/key-value-cache.ts:53:20)
> tmp-muse-1  |     at async run (file:///usr/app/node_modules/p-queue/dist/index.js:109:36)
> tmp-muse-1  |
> tmp-muse-1  | /********************************************************************************************************************************************************************************************************
> tmp-muse-1  | failed at func parseItem: type lockupViewModel is not known
> tmp-muse-1  | pls post the the files in /usr/app/node_modules/ytsr/dumps to https://github.com/TimeForANinja/node-ytsr/issues
> tmp-muse-1  | os: linux-x64, node.js: v18.20.4, ytsr: 3.8.4
> ```
> 
> I've just pulled the updates from [3c161b7](https://github.com/codetheweb/muse/commit/3c161b77dd2d8148e7821d470ae550626a24b196) and will test today.

I think this isn't an issue caused by this PR (see #1000). It's an error with ytsr (deprecated module for doing YT search requests). There is also an open PR for replacing it with Distube's fork #1024.

**Comment by EugeneLeclerc** [comment]
> I think this isn't an issue caused by this PR (see #1000). It's an error with ytsr (deprecated module for doing YT search requests). There is also an open PR for replacing it with Distube's fork #1024.

Yeah this PR cannot and should not fix everything. The 403 errors are fixed, thats already a good enough reason to merge as-is.

**Review by codetheweb — APPROVED** [review]
sorry for the delay, thank you for the patch!

**Pull request merged**